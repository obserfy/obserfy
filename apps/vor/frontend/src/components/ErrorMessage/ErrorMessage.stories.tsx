import { FC } from "react"
import ErrorMessage from "./ErrorMessage"

export default {
  title: "Core/ErrorMessage",
  component: ErrorMessage,
  parameters: {
    componentSubtitle: "Just a simple ErrorMessage",
  },
}

export const Basic: FC = () => (
  <ErrorMessage error={new Error("Are you mad?")} />
)
