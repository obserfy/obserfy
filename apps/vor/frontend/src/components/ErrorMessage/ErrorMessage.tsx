import { FC } from "react"
import { TextProps } from "theme-ui"
import { Typography } from "../Typography/Typography"

interface Props extends TextProps {
  error: unknown
}
export const ErrorMessage: FC<Props> = ({ error, ...props }) => (
  <Typography.Body {...props} sx={{ textAlign: "center" }} color="error">
    {error instanceof Error && error.message}
  </Typography.Body>
)

export default ErrorMessage
