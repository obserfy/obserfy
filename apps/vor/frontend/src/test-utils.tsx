/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
