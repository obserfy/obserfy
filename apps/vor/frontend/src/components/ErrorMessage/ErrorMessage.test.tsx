import { render } from "../../test-utils"
import { Basic } from "./ErrorMessage.stories"

describe("ErrorMessage", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
