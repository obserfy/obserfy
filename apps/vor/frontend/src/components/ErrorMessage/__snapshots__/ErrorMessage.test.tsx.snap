// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  color: var(--theme-ui-colors-error);
  text-align: center;
}

<div>
  <div
    class="emotion-0"
  >
    <p
      class="emotion-1"
    >
      Are you mad?
    </p>
  </div>
</div>
`;
